pkg/snapstore/s3_snapstore.go

```diff
@@ -275,9 +280,13 @@ func NewS3FromClient(bucket, prefix, tempDir string, maxParallelChunkUploads uin
 
 // Fetch should open reader for the snapshot file from store
 func (s *S3SnapStore) Fetch(snap brtypes.Snapshot) (io.ReadCloser, error) {
+	sseCreds := getSSECreds()
```

Comment:
`getSSECreds()` performs a `LookupEnv()` for the SSE Credentials, calculates the MD5 sum and encodes it to a string to create a new instance of the struct `sseCredentials`.
This function is invoked in every `S3SnapStore` method, for every invocation - and these methods are invoked quite frequently.
Environment lookup followed by the MD5 sum calculation would be more resource intensive than a simple memory lookup for the store value.

Also, fetching credentials should not be under the purview of `S3Snapstore` methods. Their task is to perform a `Fetch`, a `Save`, etc.
Do you expect the SSE Credentials to change dynamically when `etcd-backup-restore` is running so that these credentials have to be looked up before every method invocation?

SSE Credentials should be a part of the `S3SnapStore` struct, or the `awsCredentials` struct. The latter would only be meaningful if SSE credentials can be passed as a part of the JSON credential file - is that possible?
The credentials are read from the environment, and are initialized as fields of the struct, which can be used for all further method invocations.

Could you please clarify?

Reply:
Thank you for your comments. I initially thought of using the `awsCredentials` struct so I can pass the SSE Key using the credentials file. But what I fail to understand is how I can access it. We pass those as `session.Options`. But for SSE, I need to be able to access it within the method, like:
```
		SSECustomerAlgorithm: &awsConfig.SSECustomerAlgorithm,
		SSECustomerKey:       &awsConfig.SSECustomerKey,
		SSECustomerKeyMD5:    &awsConfig.SSECustomerKeyMD5,
```
So I went with the env variable approach. If there is a way, I'll definitely want to avoid env vars and just use the credentials file.

Reply:
For now, using env var, I've modified the code to do a `EnvLookUp` once per client invocation like the other creds. And we are accessing it via the `S3SnapStore` struct

Reply:
Resolving this since the credentials will be read from the configuration files from the current code.